File: telerik_ui_for_winforms_courseware_chm/Courseware/Grid/CS/RadGridView/HierarcyAuto/RadGridViewLab6.cs
using System;
using System.Windows.Forms;

namespace _06_HierarcyAuto
{
    public partial class RadGridViewLab6 : Form
    {
        public RadGridViewLab6()
        {
            InitializeComponent();
        }

        private void RadGridViewLab6_Load(object sender, EventArgs e)
        {
            AdventureWorksDataSetTableAdapters.VendorTableAdapter vendorTA =
              new AdventureWorksDataSetTableAdapters.VendorTableAdapter();
            AdventureWorksDataSetTableAdapters.ProductVendorTableAdapter productvendorTA =
              new AdventureWorksDataSetTableAdapters.ProductVendorTableAdapter();
            vendorTA.Fill(adventureWorksDataSet.Vendor);
            productvendorTA.Fill(adventureWorksDataSet.ProductVendor);

            radGridView1.AutoGenerateHierarchy = true;
            radGridView1.DataMember = "Vendor";
            radGridView1.MasterTemplate.BestFitColumns();
            radGridView1.MasterTemplate.Templates[0].BestFitColumns();
        }
    }
}

File: telerik_ui_for_winforms_courseware_chm/Courseware/Grid/VB/RadGridView/HierarcyAuto/RadGridViewLab6.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms

Namespace _06_HierarcyAuto
  Public Partial Class RadGridViewLab6
	  Inherits Form
	Public Sub New()
	  InitializeComponent()
	End Sub

	Private Sub RadGridViewLab6_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
	  Dim vendorTA As AdventureWorksDataSetTableAdapters.VendorTableAdapter = New AdventureWorksDataSetTableAdapters.VendorTableAdapter()
	  Dim productvendorTA As AdventureWorksDataSetTableAdapters.ProductVendorTableAdapter = New AdventureWorksDataSetTableAdapters.ProductVendorTableAdapter()
	  vendorTA.Fill(adventureWorksDataSet.Vendor)
            productvendorTA.Fill(adventureWorksDataSet.ProductVendor)

            radGridView1.AutoGenerateHierarchy = True
            radGridView1.DataMember = "Vendor"
            radGridView1.MasterTemplate.BestFitColumns()
            radGridView1.MasterTemplate.Templates(0).BestFitColumns()
	End Sub
  End Class
End Namespace
